packages/react-devtools-shared/src/inspectedElementCache.js

```diff
@@ -202,15 +203,86 @@ export function checkForUpdate({
         });
       }
     },
-
-    // There isn't much to do about errors in this case,
-    // but we should at least log them so they aren't silent.
-    error => {
-      console.error(error);
-    },
   );
 }
 
+function createPromiseWhichResolvesInOneSecond() {
+  return new Promise(resolve => setTimeout(resolve, 1000));
+}
+
+export function startElementUpdatesPolling({
+  bridge,
+  element,
+  refresh,
+  store,
+}: {
+  bridge: FrontendBridge,
+  element: Element,
+  refresh: RefreshFunction,
+  store: Store,
+}): {abort: () => void, pause: () => void, resume: () => void} {
+  let aborted = false;
+  let paused = false;
+  let running = false;
+
+  function abort() {
+    aborted = true;
+  }
+
+  function resume() {
+    if (running) {
+      return;
+    }
+
+    paused = false;
+    poll();
+  }
+
+  function pause() {
+    if (paused) {
+      return;
+    }
+
+    running = false;
+    paused = true;
+  }
+
+  function poll(): Promise<void> {
+    running = true;
+
+    return Promise.allSettled([
+      checkForUpdate({bridge, element, refresh, store}),
+      createPromiseWhichResolvesInOneSecond(),
```

Comment:
feel free to ignore, don't feel too strongly: should this per second throttling be part of this or the caller? it's not entirely clear this function does this

Reply:
The idea here is to throttle `checkForUpdate` calls: no more than once in 1 second.

Still, if Promise returned by `checkForUpdate` takes more than 1 second to resolve, `Promise.allSettled` will only resolve once all Promises are settled, so its basically a safety-check that we would not spam with errors if `checkForUpdate` instantly returns rejected Promise

Also, if `checkForUpdate`'s Promise will resolve in >1 second, we would not add this 1 second to run next poll request